test/e2e/UniswapTrade.t.sol

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8;
+
+import {Vm} from "forge-std/Vm.sol";
+
+import {IERC20} from "src/contracts/interfaces/IERC20.sol";
+
+import {GPv2Interaction} from "src/contracts/libraries/GPv2Interaction.sol";
+import {GPv2Order} from "src/contracts/libraries/GPv2Order.sol";
+import {GPv2Signing} from "src/contracts/mixins/GPv2Signing.sol";
+
+import {SettlementEncoder} from "../libraries/encoders/SettlementEncoder.sol";
+import {Registry, TokenRegistry} from "../libraries/encoders/TokenRegistry.sol";
+import {Helper, IERC20Mintable} from "./Helper.sol";
+
+using SettlementEncoder for SettlementEncoder.State;
+using TokenRegistry for TokenRegistry.State;
+using TokenRegistry for Registry;
+
+interface IUniswapV2Factory {
+    function createPair(address, address) external returns (address);
+}
+
+interface IUniswapV2Pair {
+    function token0() external view returns (address);
+    function mint(address) external;
+    function swap(uint256, uint256, address, bytes calldata) external;
+}
+
+contract UniswapTradeTest is Helper(true) {
+    IERC20Mintable dai;
+    IERC20Mintable wETH;
+
+    IUniswapV2Factory factory;
+    IUniswapV2Pair uniswapPair;
+
+    bool isWethToken0;
+
+    function setUp() public override {
+        super.setUp();
+
+        dai = deployMintableErc20("dai", "dai");
+        wETH = deployMintableErc20("wETH", "wETH");
+
+        factory = IUniswapV2Factory(0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f);
+        uniswapPair = IUniswapV2Pair(factory.createPair(address(wETH), address(dai)));
+
+        isWethToken0 = uniswapPair.token0() == address(wETH);
```

Comment:
This is missing a comment from the test being migrated from.

Reply:
doesn't apply